c++/src/kj/memory.h

```diff
@@ -574,15 +574,13 @@ const HeapDisposer<T> HeapDisposer<T>::instance = HeapDisposer<T>();
 template <typename T, void(*F)(T*)>
 class CustomDisposer: public Disposer {
 public:
-  static const CustomDisposer instance;
```

Comment:
Could this have been declared `constexpr` in-place, without un-nesting it? Not that I really care, more a curiosity.

Reply:
Sadly, no, because the type is still incomplete in the compiler's eyes at that point. I scratched my head a bit trying to think around it, then gave up.